workflow-templates/build.yml

```diff
@@ -0,0 +1,122 @@
+name: Build
+run-name: "Branch Build (${{ github.run_attempt }}.${{ github.run_number }}) - ${{ github.ref_name }}"
+
```

Comment:
Add shell defaults since this is an entry point:

```yaml
defaults:
  run:
    shell: bash
```

And we don't add concurrency limits here because we allow multiple concurrent branch builds, right? I believe that makes sense, only release and PR builds should kill any previous running builds.